couchbase/query_index_manager.hxx

```diff
@@ -88,7 +88,7 @@ public:
    * @since 1.0.0
    * @committed
    */
-  void create_index(std::string bucket_name,
+  void create_index(const std::string& bucket_name,
                     std::string index_name,
                     std::vector<std::string> keys,
```

Comment:
Should the rest of the parameters here and in the methods below also become const references?

Reply:
Probably they should. While taking by value and moving it later looks more efficient, we still transform those keys into JSON at some point, and treat them as temporary values anyway, so it is kind of waste of the memory. 

So in this context, taking it by reference would be more reasonable, as we do not want to actually use this chunk of memory for a long time (as a part of request), but rather convert it into some internal representation in some front-end.